File: JS Exams/01.  Basics Exam Collection/Basics Online Exam - 20 and 21 April 2019/04. Easter Shop/04. Easter Shop.js
function easterShop(...arg) {

    let allEggs = +arg.shift()
    let totalBoughtEggs = 0

    while (true) {
        command = arg.shift()
        if (command === 'Close')
            return `Store is closed!\n${totalBoughtEggs} eggs sold.`

        let eggsCount = +arg.shift()

        if (command === 'Buy') {
            if (allEggs >= eggsCount) {
                allEggs -= eggsCount
                totalBoughtEggs += eggsCount
            } else {
                return `Not enough eggs in store!\nYou can buy only ${allEggs}.`
            }
        } else if (command === 'Fill') {
            allEggs += eggsCount
        }
    }
}

//console.log(easterShop(13, 'Buy', 8, 'Fill', 3, 'Buy', 10))

File: Python Exams/01. Basic Exam Collection/Basics Online Exam - 20 and 21 April 2019/04. Easter Shop/04. Easter Shop.py
all_eggs = int(input())
total_bought_eggs = 0

while True:
    command = input()
    if command == 'Close':
        print(f'Store is closed!\n{total_bought_eggs} eggs sold.')
        break

    eggs_count = int(input())

    if command == 'Buy':
        if all_eggs >= eggs_count:
            all_eggs -= eggs_count
            total_bought_eggs += eggs_count
        else:
            print(f'Not enough eggs in store!\nYou can buy only {all_eggs}.')
            break
    elif command == 'Fill':
        all_eggs += eggs_count


